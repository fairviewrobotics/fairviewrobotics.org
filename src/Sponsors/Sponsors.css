/* TODO: link to page css */
.page-title {
  position: absolute;
  font-size: 10vw;
  font-weight: 100;
  text-align: center;
  display: block;
  left: 0;
  right: 0;
  top: calc(5vw + 50px);
  color: #000;
}