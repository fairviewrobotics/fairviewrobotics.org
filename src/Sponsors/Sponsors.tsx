import React, { PureComponent } from 'react';
import SponsorList from "./SponsorList/SponsorList";

import intro11 from '../images/intro/11.jpg';
import Page from "../Page/Page";

export type SponsorsProps = {
  sponsors: {
    name: string;
    size: number;
    url: string;
    src: string;
  }[];
}

export default class Sponsors extends PureComponent<SponsorsProps> {

  componentDidMount() {
    window.scrollTo(0, 0);
  }

  render() {
    return (
      <Page
        backgroundSrc={intro11}
        title="Supporters"
      >
        <SponsorList sponsors={this.props.sponsors}/>
      </Page>
    );
  }
}
