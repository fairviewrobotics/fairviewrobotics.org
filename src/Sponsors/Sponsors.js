import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import SponsorList from "./SponsorList/SponsorList";
import BackgroundImage from "../BackgroundImage/BackgroundImage";

import './Sponsors.css';

import intro11 from '../images/intro/11.jpg';

export default class Sponsors extends PureComponent {

  static propTypes = {
    sponsors: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      size: PropTypes.number.isRequired,
      url: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired
    })).isRequired
  };

  render() {
    return (
      <div>
        <BackgroundImage src={intro11}/>
        <h1 className="page-title">SPONSORS</h1>
        <SponsorList sponsors={this.props.sponsors}/>
      </div>
    );
  }
}
