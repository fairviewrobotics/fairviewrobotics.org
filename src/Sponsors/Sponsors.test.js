import React from "react";
import { shallow } from 'enzyme';
import Sponsors from "./Sponsors";

describe('Sponsors', () => {
  let wrapper;

  it('renders without crashing', () => {
    const sponsors = [];

    shallow(<Sponsors sponsors={sponsors}/>);
  });

  it('should scroll up on load', () => {
    jest.spyOn(global.window, 'scrollTo');

    shallow(<Sponsors sponsors={[]}/>);

    expect(global.window.scrollTo).toHaveBeenCalledWith(0, 0);
  });

  describe('given `sponsors`', () => {
    beforeEach(() => {
      const sponsors = [
        {
          name: 'Cheez it',
          size: 90,
          url: 'http://www.urltocheese.com/',
          src: 'http://www.cheeseimage.com/'
        },
        {
          name: 'Mcdonalds',
          size: 30,
          url: 'http://www.urltomd.com/',
          src: 'http://www.mdimage.com/'
        },
        {
          name: 'Chef Boyardee',
          size: 50,
          url: 'http://www.urltoboyardee.com/',
          src: 'http://www.boyardeeimage.com/'
        }
      ];

      wrapper = shallow(
        <Sponsors sponsors={sponsors}/>
      );
    });

    it('should render a page with a list of sponsors', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});