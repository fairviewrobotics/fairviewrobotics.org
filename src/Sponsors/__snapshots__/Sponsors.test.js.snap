// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sponsors given \`sponsors\` should render a page with a list of sponsors 1`] = `
<Page
  backgroundSrc="11.jpg"
  title="sponsors"
>
  <SponsorList
    sponsors={
      Array [
        Object {
          "name": "Cheez it",
          "size": 90,
          "src": "http://www.cheeseimage.com/",
          "url": "http://www.urltocheese.com/",
        },
        Object {
          "name": "Mcdonalds",
          "size": 30,
          "src": "http://www.mdimage.com/",
          "url": "http://www.urltomd.com/",
        },
        Object {
          "name": "Chef Boyardee",
          "size": 50,
          "src": "http://www.boyardeeimage.com/",
          "url": "http://www.urltoboyardee.com/",
        },
      ]
    }
  />
</Page>
`;
